osu.Game/Rulesets/Objects/Types/IHasHitsoundTiming.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Game.Rulesets.Objects.Types
+{
+    /// <summary>
+    /// A HitObject which has its hitsound at a specific time along its duration. Will be used for editor timeline display.
+    /// </summary>
+    public interface IHasHitsoundTiming
+    {
+        /// <summary>
+        /// The current hitsound timing of this hit object.
+        /// 0 means at the start of the object, 1 means at the end of the object.
+        /// </summary>
+        float HitsoundTiming { get; }
```

Comment:
This in general being [0;1]-ranged is a bit baffling too. It should be an absolute time instant IMO (so `.GetEndTime()` in this case).

Reply:
I kinda dislike how complex the logic has gotten after your suggestion, but it does make sense to not over-specialise it to the `TimelineHitObjectBlueprint` usecase.